server/pbench/bin/pbench-verify-backup-tarballs

```diff
@@ -0,0 +1,128 @@
+#! /bin/bash
+
+# Usage: pbench-verify-backup-tarballs <primary> <backup> <pbench> [ <pbench-local> ]
+
+# Method: in each directory, concatenate all the .md5 files and
+# pass them to md5sum for verification (eliminate DUPLICATEs for now: they
+# are handled incorrectly by pbench-{move,copy}-results and that needs to
+# be fixed).
+
+# For each directory, select all the files that don't pass the md5 check
+# and report them. TBD: Report also whether the two sets match between directories
+# (IOW, if they were corrupted to begin with).
+
+# Then compare the rest: there might be a few more files in the primary dir
+# than in the backup dir, if any additional tarballs have been sent and the
+# backup has not dealt with them yet. That is reported but it's only temporary
+# so it's OK.
+
+# Any other discrepancy is flagged and reported.
+
+# The first two arguments are specific to this script. The last two are
+# consumed by pbench-base.sh
+
+primary=$1
+backup=$2
+shift 2
+
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+if [[ -z "$1" ]]; then
+    echo "$PROG: Missing source pbench directory argument" >&2
+    exit 1
+fi
+TOP="$1"
+TOP_LOCAL="$2"
+
+. $dir/pbench-base.sh
+
+# work files
+listp=$TMP/$PROG/primary.$$
+listb=$TMP/$PROG/backup.$$
+tmp=$TMP/$PROG/tmp.$$
+ponly=$TMP/$PROG/primary.only
+bonly=$TMP/$PROG/backup.only
+report=$TMP/$PROG/report.$$
+
+# make sure the directory exists
+mkdir -p $TMP/$PROG
+
+trap "rm -f $listp $listp.failed $listb $listb.failed $tmp $ponly $bonly $report; rmdir $TMP/$PROG" EXIT INT QUIT
+
+# exit quickly if something's wrong
+cd $primary || exit 1
+cd $backup || exit 2
+
+if [ -z "$mail_recipients" ] ;then
+    exit 3
+fi
+
+# primary location
+cd $primary || exit 1
+for d in $(find . -maxdepth 1 -type d) ;do
+     pushd $d > /dev/null || exit 3
+     files=$(ls | grep md5 | grep -v DUPLICATE__NAME)
+     if [ ! -z "${files}" ] ;then
+         md5sum -c ${files} | sed 's;^;'$d/';'
+     fi 
+     popd >/dev/null
+done | sort > $listp
+
+# backup location
+cd $backup || exit 1
+for d in $(find . -maxdepth 1 -type d) ;do
+     pushd $d > /dev/null || exit 3
+     files=$(ls | grep md5 | grep -v DUPLICATE__NAME)
```

Comment:
It could, but it's unlikely: this is a list of the top level directory only, i.e. the list of controllers, which is unlikely to get much bigger than it is already. OTOH, better safe than sorry: I'll change it to  a "while read".